src/wazuh_db/wdb.c

```diff
@@ -494,38 +495,44 @@ int wdb_create_agent_db2(const char * agent_id) {
     }
 
     snprintf(path, OS_FLSIZE, "%s/%s.db", WDB2_DIR, agent_id);
+    snprintf(path_temp, OS_FLSIZE, "%s.new", path);
 
-    if (!(dest = wfopen(path, "w"))) {
+    if (!(dest = wfopen(path_temp, "w"))) {
         merror("Couldn't create database '%s': %s (%d)", path, strerror(errno), errno);
         fclose(source);
         return -1;
     }
 
     while (nbytes = fread(buffer, 1, 4096, source), nbytes) {
         if (fwrite(buffer, 1, nbytes, dest) != nbytes) {
-            unlink(path);
+            unlink(path_temp);
             result = -1;
             break;
         }
     }
 
     fclose(source);
     if (fclose(dest) == -1) {
-        merror("Couldn't create file %s completely ", path);
+        merror("Couldn't create file %s completely", path_temp);
         return -1;
     }
 
     if (result < 0) {
-        unlink(path);
+        unlink(path_temp);
```

Comment:
@LucioDonda this means a writing issue. This would already print an error when calling `wdb_create_agent_db2()`:

https://github.com/wazuh/wazuh/blob/c0f187638fb85fb8c4e661fa1f576401c229df2c/src/wazuh_db/wdb.c#L414-L415